### PR TITLE
Colin/feat 127 custom node

### LINKED ISSUE #127: Implement custom Node with Input/Edit functionality
### Acceptance criteria
- Implement custom nodes according to figma
- Users can update existing nodes data(content) through double clicking the node.

### PR DESCRIPTION
**Pull request by colinvandergraaf**
This PR introduces a new React Flow component and  LinktaNode component to be used as our default Nodes for Linkta Flows as per  the [figma](https://www.figma.com/design/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?node-id=2564-13310&t=0PIjyV8A7wfGJs6e-0)

Note: Current branch still has placeholder Flow functionality 
Fixes #127 

## Type of change

- [x] ✨  New feature (non-breaking change which adds functionality)
Screenshot 2024-06-04 at 10 24 32 PM

Screenshot 2024-06-04 at 10 23 40 PM

## How Has This Been Tested?
Smoke tested in the browser. Placeholder data is still in component

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: client/features/output-visualization-page/components/Flow.tsx
Comment (kinskiwu) [inline client/features/output-visualization-page/components/Flow.tsx:1]: Great job on the component! Seeing typescript is screaming a bit in this file, maybe we can update the types for nodes and edges to align with ReactFlow's expected types.

**Review by kinskiwu — COMMENTED** [review]
Good work @colinvandergraaf. Good to go after we address the typing in Flow.

**Review by ChTiSh — APPROVED** [review]
Thank you @colinvandergraaf, probably can get this merged so that #126 can use/adjust

File: client/features/output-visualization-page/components/Flow.tsx
Comment (colinvandergraaf) [reply client/features/output-visualization-page/components/Flow.tsx:1]: thanks for catching this!

**Review by kinskiwu — APPROVED** [review]

**Pull request merged**